lib/pages/chat/chat_app_bar_title.dart

```diff
@@ -113,37 +142,49 @@ class ChatAppBarTitle extends StatelessWidget {
     );
   }
 
-  _buildStatusContent(BuildContext context, Room room) {
+  bool hasNoConnection(SyncStatusUpdate status) {
+    return status.status == SyncStatus.error &&
+        (status.error?.exception is SocketException ||
+            status.error?.exception is MatrixConnectionException);
```

Comment:
why not use `[connection_pls](https://pub.dev/packages/connectivity_plus)`? it will be a passive listener.

Reply:
right here we need to depend on the `sync update event`? How the `sync` request work? Repeat request to `sync`? What is the interval?

Reply:
Moreover, how about reuse it in other screen?